### PR TITLE
refactor: migrating from class to functional components

### LINKED ISSUE #3: migration to functional components
Migration of the current Class Component into a Functional component.

This would affect the exported Component and the README docs.

### PR DESCRIPTION
**Pull request by SeanCassiere**
closes #3

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,7 +41,6 @@
   },
   "dependencies": {
     "credit-card-type": "^8.3.0",
-    "eslint-plugin-react-hooks": "^4.3.0",
```
Comment (SeanCassiere) [inline package.json:44]: this should be dev dependancy

Comment (SeanCassiere) [reply package.json:44]: nevermind, it turns out I've moved it to dev dependencies and this was an artifact of a deleted branch

File: README.md [hunk 2]
```diff
@@ -1,75 +1,76 @@
...
+      <Cards
+        number={state.number}
+        expiry={state.expiry}
+        cvc={state.cvc}
+        name={state.name}
+        focused={state.focus}
+      />
+      <form>
+        <input
+          type="tel"
```
Comment (felquis) [inline README.md:55]: should be `type="number"`

**Review by felquis — CHANGES_REQUESTED** [review]
Looking good, great work.

File: README.md (hunk 2 above)
Comment (SeanCassiere) [reply README.md]: fixed

**Review by felquis — APPROVED** [review]

**Pull request merged**